### DISCUSSION TITLE
Proxy seacrh latency is good , but app SDK latency is high

### DISCUSSION THREAD

**Discussion by sumanthhuddar** [milvus-io/milvus#34573] [Q&A and General discussion] [answered]
We are perf tetsing milvus 2.4.5 , distributed set up. data set ~5M. Index : HNSW, M=16, EF construction 100. with seacrh parameter k=100 , and 10K QPS we are seeing good latency at SDK(measured at app), and Milvus proxy seacrh latency at 30ms. When we increase k to 250, proxy seacrh latency increses to 40-60ms , but app side seacrh latency spikes to seconds. Anyone have experienced it ?? What may be the cause here and what parameters we  need to tune. If "proxy search latency" is good , does that mean everything is good from Milvus , or is there anything need to be tuned at Milvus proxy nodes.?

**Comment by yhmo** [comment, marked as answer]
topk=100 change to topk=250, the difference is: more id-distance pairs are returned by the proxy node.
Each id-distance pair size is 12 bytes. With 10K qps and topk100, the returned id-distance pairs per second is 10K * 100 * 12 bytes = 11.5MB. With 10K qps and topk250, the returned size per second is 28.6MB. I think this should not be a problem for network bandwidth. But if you set the output_fields to return more data, the situation might be different.

**Comment by xiaofan-luan** [comment]
analyze on:
proxy cpu and proxy queue time, add more proxy might help on this use case.
if you have output field, check io and network bandwidth.
